packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/TypeFactory.cs

```diff
@@ -298,6 +298,9 @@ protected virtual ParameterProvider CreateParameterCore(InputParameter parameter
                     BytesKnownEncoding.Base64Url => SerializationFormat.Bytes_Base64Url,
                     _ => throw new IndexOutOfRangeException($"unknown encode {primitiveType.Encode}")
                 },
+                InputPrimitiveTypeKind.Integer or InputPrimitiveTypeKind.Int8 or InputPrimitiveTypeKind.Int16 or InputPrimitiveTypeKind.Int32
```

Comment:
if `@encode("string")` is added to other types, nothing will happen. Is there any warning for the usage?

Reply:
I tend to let decorator or linter to do the validation, v.s. csharp generator validates it.